src/lib/menu/menu-trigger.ts

```diff
@@ -197,12 +197,15 @@ export class MdMenuTrigger implements AfterViewInit, OnDestroy {
    */
   private _getPosition(): ConnectedPositionStrategy  {
     const positionX: HorizontalConnectionPos = this.menu.positionX === 'before' ? 'end' : 'start';
-    const positionY: VerticalConnectionPos = this.menu.positionY === 'above' ? 'bottom' : 'top';
-
+    const overlayY: VerticalConnectionPos = this.menu.positionY === 'above' ? 'bottom' : 'top';
+    let originY = overlayY;
+    if (this.menu.overlayTrigger) {
+      originY = overlayY === 'top' ? 'bottom' : 'top';
+    }
```

Comment:
One would think so. However I believe the current implementation produces the desired effect. Are you able to clone the pr and run the demo-app and verify? I have done so and I believe the functionality is accurate.